jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/MultiPartServletTest.java

```diff
@@ -371,6 +391,152 @@ protected void service(HttpServletRequest request, HttpServletResponse response1
         assertThat(fileList.length, is(0));
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testDefaultTempDirectory(boolean eager) throws Exception
+    {
+        start(new HttpServlet()
+        {
+            @Override
+            protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
+            {
+                Collection<Part> parts = request.getParts();
+                assertNotNull(parts);
+                assertEquals(1, parts.size());
+                Part part = parts.iterator().next();
+                assertEquals("part1", part.getName());
+                Collection<String> headerNames = part.getHeaderNames();
+                assertNotNull(headerNames);
+                assertEquals(2, headerNames.size());
+                String content1 = IO.toString(part.getInputStream(), UTF_8);
+                assertEquals("content1", content1);
+            }
+        }, new MultipartConfigElement(null, MAX_FILE_SIZE, -1, 0), eager);
+
+        try (Socket socket = new Socket("localhost", connector.getLocalPort()))
+        {
+            OutputStream output = socket.getOutputStream();
+
+            String content = """
+                --A1B2C3
+                Content-Disposition: form-data; name="part1"
+                Content-Type: text/plain; charset="UTF-8"
+                                
+                content1
+                --A1B2C3--
+                """;
+            String header = """
+                POST / HTTP/1.1
+                Host: localhost
+                Content-Type: multipart/form-data; boundary="A1B2C3"
+                Content-Length: $L
+                                
+                """.replace("$L", String.valueOf(content.length()));
+
+            output.write(header.getBytes(UTF_8));
+            output.write(content.getBytes(UTF_8));
+            output.flush();
+
+            HttpTester.Response response = HttpTester.parseResponse(socket.getInputStream());
+            assertNotNull(response);
+            assertEquals(HttpStatus.OK_200, response.getStatus());
+        }
+    }
+
+    @Test
+    public void testDefaultTempDirectoryWithMultipleContexts() throws Exception
+    {
+        List<File> tmpDirs = new CopyOnWriteArrayList<>();
+        HttpServlet servlet = new HttpServlet()
+        {
+            @Override
+            protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
+            {
+                File tmpDirFromAttribute = (File)request.getAttribute(ServletContext.TEMPDIR);
```

Comment:
Using `request.getAttribute(ServletContext.TEMPDIR)` should return null.
That attribute doesn't exist on the request, it only exists on the context.

This code should be using `getServletContext().getAttribute(ServletContext.TEMPDIR)` instead.

Reply:
You're right, let me fix that.